File: swift/Neetcode/Arrays/Easy/58.LengthLastWord.swift
// Given a string s consisting of words and spaces, 
// return the length of the last word in the string.

func lengthOfLastWord(_ s: String) -> Int {
    var length = 0
    var foundLetter = false

    for letter in s.reversed() {
        if letter != " " {
            foundLetter = true
            length += 1
        } else if foundLetter {
            return length
        }
    }

    return length
}

print(lengthOfLastWord("Hello World"))
print(lengthOfLastWord("   fly me   to   the moon  "))
print(lengthOfLastWord("luffy is still joyboy"))

File: python/Neetcode/Arrays/Easy/58.LengthLastWord.py
#Given a string s consisting of words and spaces, 
#return the length of the last word in the string.


def lengthOfLastWord(s):

    i, word_length = len(s) - 1, 0

    while s[i] == ' ':
        i -= 1
    while s[i] != ' ' and i >= 0:
        word_length += 1
        i -= 1

    return word_length

print(lengthOfLastWord("Hello World"))
print(lengthOfLastWord("   fly me   to   the moon  "))
print(lengthOfLastWord("luffy is still joyboy"))